src/Xamarin.Android.Tools.Aidl/CSharpCodeGenerator.cs

```diff
@@ -277,6 +277,20 @@ protected override bool OnTransact (int code, global::Android.OS.Parcel data, gl
 					w.WriteLine ("\t\t\t\t{0} {1} = default ({0});", ToOutputTypeName (name_cache.ToCSharp (a.Type)), "arg" + i);
 					if (a.Modifier == null || a.Modifier.Contains ("in"))
 						w.WriteLine ("\t\t\t\t{0}", GetCreateStatements (a.Type, "data", "arg" + i));
+					else if (a.Modifier != null && a.Modifier.Contains ("out")) {
+						if (a.Type.ArrayDimension > 0) {
```

Comment:
This code block appears to only support `a.Type.ArrayDimension` being `0` (no array)` or `1` (single-dimension array).  Could it also be `2` or more, e.g. for rectangular arrays such as `int[][]`?

It may be "safer" to `switch (a.Type.ArrayDimension)` and e.g. throw if it's an unsupported value.

Reply:
Answer: no, AIDL doesn't (currently) support mult-dimensional arrays.